grapl-security/grapl
# Fix the React proxy settings to point to our web-ui backend

### Which issue does this PR correspond to?

Not sure if filed, but the development loop for frontend (engagement-view) was essentially broken per standard processes for React development. TIL.

### What changes does this PR make to Grapl? Why?

This configures the React middleware proxy to point to the backend web server in our local deployments (grapl-web-ui) for our auth and graphql apis, allowing frontend devs to run `yarn start` for development.

1. Fix the React settings to point to our backend web server.
2. Add a Make target to the engagement-view Makefile for running `yarn start` in a dev container.

### How were these changes tested?

1. `$ KEEP_TEST_ENV=1 make test-e2e`
2. `$ cd src/js/engagement_view`
3. `$ make start`
4. Browse to http://localhost:3000
5. login with test user credentials
6. See the engagement from the E2E tests
7. Make a simple color change to one of the component things (I forget which), save the file.
8. Observed the color updated in the browser without any other input.
